src/Parse/EnvFile.php

```diff
@@ -0,0 +1,238 @@
+<?php namespace  Winter\Storm\Parse;
+
+use Winter\Storm\Support\Str;
+use Winter\Storm\Parse\Contracts\DataFileInterface;
+
+/**
+ * Class EnvFile
+ */
+class EnvFile implements DataFileInterface
+{
+    /**
+     * @var array Lines of env data
+     */
+    protected $env = [];
+
+    /**
+     * @var array Map of variable names to line indexes
+     */
+    protected $map = [];
+
+    /**
+     * @var string|null Filepath currently being worked on
+     */
+    protected $filePath = null;
+
+    /**
+     * EnvFile constructor
+     */
+    public function __construct(string $filePath)
```

Comment:
@jaxwilko discovered this using PHPStan, but if you don't make the constructor API concrete, it can make creating static instances a problem if people extend the class and change up the API. Simple fix is to make the constructor final if you don't intend for people to change the constructor, even after extending.

```suggestion
    final public function __construct(string $filePath)
```